File: Initial_Python_Version_Backend/library/chat_status.py
import time

class ChatStatus:
    def __init__(self):
        self.is_away = False
        self.voice_on = False
        self.current_mood = "Neutral"
        self.chat_duration = 0
        self.last_user_input_time = 0
        self.last_user_input_change_time = 0

    def user_input_is_given(self):
        self.last_user_input_time = time.time()

    def user_input_is_changed(self):
        self.last_user_input_change_time = time.time()

    def get_last_input_time_diff(self):
        return time.time() - self.last_user_input_time

    def get_last_input_change_time_diff(self):
        return time.time() - self.last_user_input_change_time

    def set_away(self, away):
        self.is_away = away

    def initialization(self):
        self.is_away = False
        self.current_mood = "Neutral"
        self.chat_duration = 0

File: React_Frontend/src/system_runner/chat_status.js
export class ChatStatus {
  constructor() {
    this.is_away = false;
    this.voice_on = false;
    this.current_mood = "Neutral";
    this.chat_duration = 0;
    this.last_user_input_time = 0;
    this.last_user_input_change_time = 0;
  }

  user_input_is_given() {
    this.last_user_input_time = Date.now();
  }

  user_input_is_changed() {
    this.last_user_input_change_time = Date.now();
  }

  get_last_input_time_diff() {
    return Date.now() - this.last_user_input_time;
  }

  get_last_input_change_time_diff() {
    return Date.now() - this.last_user_input_change_time;
  }

  set_away(away) {
    this.is_away = away;
  }

  initialization() {
    this.is_away = false;
    this.current_mood = "Neutral";
    this.chat_duration = 0;
  }
}